### PR TITLE
GH#175 Test Spec Document - FPD L3 

### PR DESCRIPTION
**Pull request by akhilbhas**
* Adding low level spec for FPD l3 test

### CODE REVIEW

File: docs/pages/ds-front-panel-display_L3_Low-Level_TestSpecification.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+## Table of Contents
+
+- [Overview](#overview)
+- [Acronyms, Terms and Abbreviations](#acronyms-terms-and-abbreviations)
+- [References](#references)
+- [Level 3 Test cases High Level Overview](#level-3-test-cases-high-level-overview)
+
+## Overview
+
+This document describes the L3 Low Level Test Specification and Procedure Documentation for the Device Settings Video Port module.
```
Comment (KarthikeyanR470) [inline docs/pages/ds-front-panel-display_L3_Low-Level_TestSpecification.md:12]: In the overview description, you have mentioned as 'Video Port'. Change that to FPD.

**Review by KarthikeyanR470 — CHANGES_REQUESTED** [review]
Added review comments

File: docs/pages/ds-front-panel-display_L3_Low-Level_TestSpecification.md (hunk 1 above)
Comment (akhilbhas) [reply docs/pages/ds-front-panel-display_L3_Low-Level_TestSpecification.md]: updated

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+### References
+
+- `High Level Test Specification` - [dsFrontPanelDevice High Level TestSpec](ds-front-panel-display_High-Level_TestSpec.md)
+- `Interface header [4.0.0]` - [dsFPD HAL header](https://github.com/rdkcentral/rdk-halif-device_settings/blob/4.0.0/include/dsFPD.h)
+
+## Level 3 Test Cases High Level Overview
+
+|#|Test-case|Description|HAL APIs|Source|Sink|
+|-|---------|-----------|--------|------|----|
+|1|Verify LED can be set On and OFF|Verify each LED can be powered on or off |`dsSetFPState()`|`Y`|`Y`|
```
Comment (Anbukannadhasan) [inline docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md:34]: How does all test cases differ from L2 test cases? How is the verification planned—manually checked by the user?

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments?

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+|2|Check Brightness Control|Verify the LED brightness control by changing the brightness |`dsSetFPBrightness()`|`Y`|`Y`|
+|3|Check LED Blink funciton|Verify the LED can be set with specified Blink rate |`dsSetFPState()`|`Y`|`Y`|
+|4|Verify LED Color support  | Verify the Front panel LED can be set to supported colors | `dsSetFPColor()` | `Y` | `Y`|
+|5|Verify All the Supported State pattern of LED | Verify the Front Panel LED can set all the pattern to be shown during different state of the device | `dsFPSetLEDState()`|`Y`|`Y`|
+
+## Level 3 Python Test Cases High Level Overview
```
Comment (hari22yuva) [inline docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md:40]: can you add this to 
Table of Contents

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+  - Specifies overall configuration for the platform
+  - Can be overridden by:
+    - Changing locally .yaml file directory
+    - Using --deviceConfig command line switch
+  - For more details refer [RAFT](https://github.com/rdkcentral/python_raft/blob/1.0.0/README.md) and [example_device_config.yml](https://github.com/rdkcentral/python_raft/blob/1.0.0/examples/configs/example_device_config.yml)
+
+- **componentProfile.yaml/platformProfile.yaml**
+  - Contains component-specific configurations
+  - Contains platform wide configuration broken down into separate components
+  - Example configuration file [dsVideoPort_Settings](https://github.com/rdkcentral/rdk-halif-test-device_settings/blob/main/profiles/sink/Sink_4K_VideoPort.yaml)
```
Comment (hari22yuva) [inline docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md:112]: please provide the front panel configuration file

**Review by hari22yuva — COMMENTED** [review]
please check review comments

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+- `High Level Test Specification` - [dsFrontPanelDevice High Level TestSpec](ds-front-panel-display_High-Level_TestSpec.md)
+- `Interface header [4.0.0]` - [dsFPD HAL header](https://github.com/rdkcentral/rdk-halif-device_settings/blob/4.0.0/include/dsFPD.h)
+
+## Level 3 Test Cases High Level Overview
+
+|#|Test-case|Description|HAL APIs|Source|Sink|
+|-|---------|-----------|--------|------|----|
+|1|Verify LED can be set On and OFF|Verify each LED can be powered on or off |`dsSetFPState()`|`Y`|`Y`|
+|2|Check Brightness Control|Verify the LED brightness control by changing the brightness |`dsSetFPBrightness()`|`Y`|`Y`|
```
Comment (srinivasgtl) [inline docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md:35]: state that all the LEDs available on the platform should be checked for brightness, color etc.

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+|3|Check LED Blink funciton|Verify the LED can be set with specified Blink rate |`dsSetFPState()`|`Y`|`Y`|
+|4|Verify LED Color support  | Verify the Front panel LED can be set to supported colors | `dsSetFPColor()` | `Y` | `Y`|
+|5|Verify All the Supported State pattern of LED | Verify the Front Panel LED can set all the pattern to be shown during different state of the device | `dsFPSetLEDState()`|`Y`|`Y`|
```
Comment (srinivasgtl) [inline docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md:38]: In description, please mention as "all the supported states on the DUT" should be validated.

Comment (akhilbhas) [reply docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md]: updated

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md (hunk 5 above)
Comment (akhilbhas) [reply docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md]: updated

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md (hunk 4 above)
Comment (akhilbhas) [reply docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md]: changed the link to FPD configuration file

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md (hunk 3 above)
Comment (akhilbhas) [reply docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md:41]: Added to table of content

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md (hunk 2 above)
Comment (akhilbhas) [reply docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md]: Updated the description.

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+  - For more details refer [RAFT](https://github.com/rdkcentral/python_raft/blob/1.0.0/README.md) and [example_device_config.yml](https://github.com/rdkcentral/python_raft/blob/1.0.0/examples/configs/example_device_config.yml)
+
+- **componentProfile.yaml/platformProfile.yaml**
+  - Contains component-specific configurations
+  - Contains platform wide configuration broken down into separate components
+  - Example configuration file [dsFPD_Settings](https://github.com/rdkcentral/rdk-halif-test-device_settings/blob/3.0.0/profiles/sink/Sink_FPD.yaml)
+
+- **testSetupConfig.yaml**
+  - This configuration file contains the list of requirements for tests to execute. Eg: Copying the streams, setting environment variables etc.
+  - Example configuration file [dsFPD_L3_testSetup.yml](../../../host/tests/dsFPD_L3_Tests/dsFPD_L3_testSetup.yml)
```
Comment (Anbukannadhasan) [inline docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md:95]: links are missing

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+- **testSuite.yaml**
+  - This configuration file contains the list of menu items for C/C++ L3 test running on `DUT`
+  - Example configuration file [dsFPD_test_suite.yml](../../../host/tests/dsClasses/dsFPD_test_suite.yml)
```
Comment (Anbukannadhasan) [inline docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md:99]: links are missing

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

**Review by Anbukannadhasan — DISMISSED** [review]
Approved

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md [hunk 9]
```diff
@@ -0,0 +1,99 @@
+# Device Settings Front Panel Display L3 Low Level Test Specification and Procedure Documentation
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Acronyms, Terms and Abbreviations](#acronyms-terms-and-abbreviations)
+- [References](#references)
+- [Level 3 Test cases High Level Overview](#level-3-test-cases-high-level-overview)
+- [Level 3 Python Test Cases High Level Overview](#level-3-python-test-casesc-high-level-overview)
```
Comment (hari22yuva) [inline docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md:9]: this link is not working

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+```mermaid
+---
+title: dsFPD - Python Class Flow
+---
+classDiagram
+    testControl <|-- ut_raft : inherits
+    class ut_raft{
+    }
+    ut_raft <|-- L3_TestClasses : inherits
+    L3_TestClasses ..> dsFPD : uses
```
Comment (hari22yuva) [inline docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md:54]: dsFPD class uses the profile 
please update that

**Review by hari22yuva — COMMENTED** [review]
Please address the review comments

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md (hunk 10 above)
Comment (akhilbhas) [reply docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md:54]: updated

File: docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md (hunk 9 above)
Comment (akhilbhas) [reply docs/pages/dsFPD/ds-front-panel-display_L3_Low-Level_TestSpecification.md]: updated

**Review by thanushreevani — APPROVED** [review]

**Review by KarthikeyanR470 — APPROVED** [review]

**Review by Anbukannadhasan — APPROVED** [review]
Approved

**Pull request merged**